### PR TITLE
[refactor] Remove using namespace <xxx> from wallet/ & util*

### PR DESCRIPTION
**Pull request by kallewoof**
This is a sub-set of #9235, including only the `wallet` dir and `util*` files.

Same-binaries check (2017-03-08):
```
$ ../bitcoin-maintainer-tools/build-for-compare.py 6996e066b538f03b8aa1f617dbb959b57ff6e727 a57845c20e96825f1256e6b7f46d67d205c859e0 --executables "src/bitcoind,src/bitcoin-cli,src/bitcoin-tx"
[...]
make[1]: Leaving directory '/tmp/repo/src'
>>> [do_build] Copying object files...
>>> [do_build] Performing basic analysis pass...
>>> [do_build] Use these commands to compare results:
>>> [do_build] $ sha256sum /tmp/compare/*.stripped
>>> [do_build] $ git diff -W --word-diff /tmp/compare/6996e066b538f03b8aa1f617dbb959b57ff6e727 /tmp/compare/a57845c20e96825f1256e6b7f46d67d205c859e0
$ sha256sum /tmp/compare/*.stripped
1bfe4311525db52b9c5d5c2b0b9ff6f8e3f3cfda0dcb1e5cd83b08b767a4a902  /tmp/compare/bitcoin-cli.6996e066b538f03b8aa1f617dbb959b57ff6e727.stripped
1bfe4311525db52b9c5d5c2b0b9ff6f8e3f3cfda0dcb1e5cd83b08b767a4a902  /tmp/compare/bitcoin-cli.a57845c20e96825f1256e6b7f46d67d205c859e0.stripped
fb0f6796776dc39e422910572a216cbad87f607cbb4920bd12d2c483cc51e231  /tmp/compare/bitcoind.6996e066b538f03b8aa1f617dbb959b57ff6e727.stripped
fb0f6796776dc39e422910572a216cbad87f607cbb4920bd12d2c483cc51e231  /tmp/compare/bitcoind.a57845c20e96825f1256e6b7f46d67d205c859e0.stripped
fe4107fcc49ba17cdfbb0edc357373ea37d833d5691af7b2367758e8b3e1d45c  /tmp/compare/bitcoin-tx.6996e066b538f03b8aa1f617dbb959b57ff6e727.stripped
fe4107fcc49ba17cdfbb0edc357373ea37d833d5691af7b2367758e8b3e1d45c  /tmp/compare/bitcoin-tx.a57845c20e96825f1256e6b7f46d67d205c859e0.stripped
$ git diff -W --word-diff /tmp/compare/6996e066b538f03b8aa1f617dbb959b57ff6e727 /tmp/compare/a57845c20e96825f1256e6b7f46d67d205c859e0
$ 
```

The affected files and namespaces:
```
util.cpp:                 std
utilmoneystr.cpp:         std
utilstrencodings.cpp:     std
utiltime.cpp:             std [unused]
wallet/db.cpp:            std
wallet/rpcdump.cpp:       std
wallet/rpcwallet.cpp:     std
wallet/test/wallet_tests.cpp: std
wallet/wallet.cpp:        std
wallet/walletdb.cpp:      std
```

### CODE REVIEW

File: src/util.cpp [hunk 1]
```diff
@@ -97,15 +97,15 @@ namespace boost {
 
 } // namespace boost
 
-using namespace std;
+// blank space for binary match
```
Comment (dcousens) [inline src/util.cpp]: ?

Comment (kallewoof) [inline src/util.cpp]: The binary check tool trips over the LOCK macros. If a line number is removed, the binary check ends up failing.

Comment (kallewoof) [inline src/util.cpp]: In hindsight I realize that comment is very unhelpful. I'll reword it.

File: src/util.cpp [hunk 2]
```diff
@@ -97,15 +97,15 @@ namespace boost {
 
 } // namespace boost
 
-using namespace std;
+
```
Comment (jtimon) [inline src/util.cpp]: Bike-shedding: you could just remove the line.

**Comment by jtimon** [comment]
utACK df8b249
Needs rebase

File: src/util.cpp (hunk 2 above)
Comment (kallewoof) [inline src/util.cpp]: Because of macros which use the line number, removing this line breaks the same-binaries-check, unfortunately.

**Comment by kallewoof** [comment]
Rebased.

**Comment by laanwj** [comment]
Needs another rebase, sorry.

**Comment by kallewoof** [comment]
@laanwj Rebased & redid same binaries check (6996e066b538f03b8aa1f617dbb959b57ff6e727 vs a57845c20e96825f1256e6b7f46d67d205c859e0). And no worries at all. I knew this might take awhile. :)

**Pull request merged**